src/public/types/table.d.ts

```diff
@@ -47,7 +47,14 @@ export interface Table<T=any, TKey=IndexableType> {
   delete(key: TKey): PromiseExtended<void>;
   clear(): PromiseExtended<void>;
   bulkGet(keys: TKey[]): PromiseExtended<T[]>;
-  bulkAdd(items: T[], keys?: IndexableTypeArrayReadonly): PromiseExtended<TKey>;
-  bulkPut(items: T[], keys?: IndexableTypeArrayReadonly): PromiseExtended<TKey>;
-  bulkDelete(keys: IndexableTypeArrayReadonly) : PromiseExtended<void>;
+
+  bulkAdd<B extends boolean>(items: T[], keys: IndexableTypeArrayReadonly, options: { allKeys: B }): B extends true ? Promise<TKey[]> : Promise<TKey>;
+  bulkAdd<B extends boolean>(items: T[], options: { allKeys: B }): B extends true ? Promise<TKey[]> : Promise<TKey>;
```

Comment:
Shouldn't the return type be `PromiseExtended` instead of a plain `Promise`?

Reply:
Whoops, I think you are right.

Reply:
Added fix in PR #980